pypeman/channels.py

```diff
@@ -808,6 +808,57 @@ async def handle(self, msg):
         return result
 
 
+class MergeChannel(BaseChannel):
+    """
+    This class permits to have multiple channel classes (watchers/listeners/..)
+    as inputs to a single channel
+    TODO: CAUTION: Not sure that input channels works with end nodes + they don't
+        appears in graph
+
+    TODO: check why we need
+        if channel.loop != self.loop:
+            channel.loop = self.loop
+    There's a bug with event loop (maybe only on tests) but weird
+    (traceback: RuntimeError: Task <Task pending name='Task-126' coro=<MergeChannel.start()
+    running at
+    /home/quentin/projects/interop2/app/vendor/common/custom_channels.py:51>
+    cb=[_run_until_complete_cb()
+    at /home/quentin/.pyenv/versions/3.10.10/lib/python3.10/asyncio/base_events.py:184]>
+    got Future
+    <Future pending cb=[_chain_future.<locals>._call_check_cancel() at
+    /home/quentin/.pyenv/versions/3.10.10/lib/python3.10/asyncio/futures.py:385]>
+    attached to a different loop
+    )
+
+    Args:
+        channels (list of channels): List of channels to use as inputs
+    """
+
+    def __init__(self, *args, chans, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.channels = chans
+        for channel in self.channels:
+            all_channels.remove(channel)
```

Comment:
why is removing the initial channel necessary. it seems to 'destroy' introspection.